ibm/service/appconfiguration/data_source_ibm_app_config_snapshot.go

```diff
@@ -0,0 +1,251 @@
+package appconfiguration
+
+import (
+	"fmt"
+	"github.com/IBM/appconfiguration-go-admin-sdk/appconfigurationv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceIBMAppConfigSnapshot() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIbmAppConfigSnapshotRead,
+
+		Schema: map[string]*schema.Schema{
+			"guid": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "GUID of the App Configuration service. Get it from the service instance credentials section of the dashboard.",
+			},
+			"git_config_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Git config id. Allowed special characters are dot ( . ), hyphen( - ), underscore ( _ ) only",
+			},
+			"git_config_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
All these arguments git_config_name, git_url, git_branch, git_file_path we are just setting the value they needs to be marked as computed not optional

Reply:
Addressed